### PR TITLE
GET_REPORTS API

### PR DESCRIPTION
**Pull request by yousiefc**
# PR Description

Added API to generate reports 

## How Has This Been Tested?
API tester client

### CODE REVIEW

File: src/api/api/BIOMD/BIOMD.js [hunk 1]
```diff
@@ -416,3 +423,171 @@ module.exports.LIST_VENDORS = async function (req, dbClient) {
 
   return res;
 };
+
+module.exports.RECORDS = async function (req, dbClient) {
```
Comment (akshaypuli) [inline src/api/api/BIOMD/BIOMD.js:427]: Could we change the API to be called GET_REPORTS

File: src/api/api/BIOMD/BIOMD.js [hunk 2]
```diff
@@ -416,3 +423,171 @@ module.exports.LIST_VENDORS = async function (req, dbClient) {
...
+    console.log(
+      (await TIMESTAMP()) +
+      `: RCU-BIOMD-I001 : RECORDS processing request packet ID: ${req.ID}`
+    );
+
+    //Copy Requester Information
+    var res = Object.assign({}, req);
+
+    //req object defintion
+    var find = Object.assign({}, req.Request.find);
```
Comment (akshaypuli) [inline src/api/api/BIOMD/BIOMD.js:440]: Could we change the schema for find? Remove the variable find and just have reportType as the top level variable?

File: src/api/api/BIOMD/BIOMD.js [hunk 3]
```diff
@@ -416,3 +423,171 @@ module.exports.LIST_VENDORS = async function (req, dbClient) {
...
+                inactive: {$sum: "$inactiveCount"},
+                totalDevices: {$sum: 1}
+              }
+            }
+          ]).toArray();
+          break;
+        }
+        case INFLOW: {
+          //The year value should also be included in the request object in find
+          var year = find.year || null;
```
Comment (akshaypuli) [inline src/api/api/BIOMD/BIOMD.js:528]: Could variable like these be pass-in through a variable called reportSpecs.year?

File: src/api/services/Communication_Link/middleware.js
Comment (akshaypuli) [inline src/api/services/Communication_Link/middleware.js:1]: Did you intend to make changes to the middleware? Looks like just spaces but want to be careful making changes to this file.

File: src/api/api/BIOMD/BIOMD.js [hunk 4]
```diff
@@ -60,6 +60,13 @@ const OPERATORS = ["eq", "in", "gt", "lt", "sb", "eq_id"];
 const DEFAULT_LIMIT_COUNT = 1000;
 const DEFAULT_SKIP_COUNT = 0;
```
Comment (akshaypuli) [inline src/api/api/BIOMD/BIOMD.js:62]: Are the following variables GLOBAL definitions or just needed by your API?

**Review by akshaypuli — CHANGES_REQUESTED** [review]
Thank You for the PR! Looks great! Just a few comments. Could you also upload the YAML test file?

**Review by oaedward — APPROVED** [review]

**Review by akshaypuli — APPROVED** [review]

**Pull request merged**